contracts/mesh-provider/src/ibc.rs

```diff
@@ -213,33 +223,110 @@ pub fn ack_list_validators(
     for val in res.validators {
         VALIDATORS.save(deps.storage, &val, &Validator::new())?;
     }
+
+    // reset retry counter
+    let mut config = CONFIG.load(deps.storage)?;
+    config.list_validators_retries_remaining = LIST_VALIDATORS_RETRIES;
+    CONFIG.save(deps.storage, &config)?;
+
     Ok(IbcBasicResponse::new())
 }
 
-pub fn fail_list_validators(_deps: DepsMut) -> Result<IbcBasicResponse, ContractError> {
-    // TODO: send another ListValidators message
-    unimplemented!();
+pub fn fail_list_validators(deps: DepsMut, env: Env) -> Result<IbcBasicResponse, ContractError> {
+    // check if we should retry
+    let cfg = CONFIG.load(deps.storage)?;
+    let channel_id = CHANNEL.load(deps.storage)?;
+    if cfg.list_validators_retries_remaining <= 0 {
+        return Err(ContractError::NoResponse("list_validators".to_string(), channel_id))
+    }
+    let mut cfg_after = cfg.clone();
```

Comment:
I see no reason to clone. I would just declare `let mut cfg = CONFIG.load()` when it is created